geomstats/geometry/lie_group.py

```diff
@@ -229,9 +230,13 @@ def exp_not_from_identity(self, tangent_vec, base_point, point_type):
 
         if point_type == "matrix":
             tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=3)
-            raise NotImplementedError()
+            lie_vec = self.compose(
+                self.inverse(base_point), tangent_vec)
+            return self.compose(
+                base_point, self.exp_from_identity(lie_vec, point_type))
 
-        raise ValueError('point_type should be \'vector\' or \'matrix\'.')
+        raise ValueError('Invalid point_type, expected \'vector\' or '
```

Comment:
Nice! I'm thinking that, at some point, we should uniformize the error messages. Right now we have various syntaxes across the codebase. I'm writing an issue.